source/common/compressor/zlib_compressor_impl.cc

```diff
@@ -27,19 +32,14 @@ void ZlibCompressorImpl::init(CompressionLevel comp_level, CompressionStrategy c
   initialized_ = true;
 }
 
-void ZlibCompressorImpl::setChunk(uint64_t chunk) { chunk_ = chunk; }
-
-void ZlibCompressorImpl::finish(Buffer::Instance& output_buffer) {
+void ZlibCompressorImpl::flush(Buffer::Instance& output_buffer) {
   process(output_buffer, Z_SYNC_FLUSH);
-  commit(output_buffer);
 }
 
+uint64_t ZlibCompressorImpl::checksum() { return zstream_ptr_->adler; }
```

Comment:
What happens if you call this before compress()?

Reply:
It works, but I added tests for both compressor and decompressor.